synapse/handlers/auth.py

```diff
@@ -45,12 +45,93 @@
 from synapse.module_api import ModuleApi
 from synapse.push.mailer import load_jinja2_templates
 from synapse.types import Requester, UserID
+from synapse.util.msisdn import phone_number_to_msisdn
 
 from ._base import BaseHandler
 
 logger = logging.getLogger(__name__)
 
 
+def client_dict_convert_legacy_fields_to_identifier(
+    submission: Dict[str, Union[str, Dict]]
+):
+    """Take a legacy-formatted login submission or User-Interactive Authentication dict and
+    updates it to feature an identifier dict instead.
+    Providing user-identifying information at the top-level of a login or UIA submission is
+    now deprecated and replaced with identifiers:
+    https://matrix.org/docs/spec/client_server/r0.6.1#identifier-types
+    Args:
+        submission: The client dict to convert. Passed by reference and modified
+    Raises:
+        SynapseError: if the dict contains a "medium" parameter that is anything other than
+            "email"
```

Comment:
This comment seems wrong? It raises if the format is invalid.

Reply:
Ah, that was a check I added, as that's technically [required by the spec for login](https://matrix.org/docs/spec/client_server/unstable#post-matrix-client-r0-login) although it's a bit of a weird requirement if `identifier` is allowed to be `m.id.phone`.

Will remove the comment though.